src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -272,6 +201,99 @@ protected void containerIsStarting(final InspectContainerResponse containerInfo,
      * @return bootstrap servers
      */
     public String getBootstrapServers() {
-        return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), this.kafkaExposedPort);
+        return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), kafkaDynamicKafkaPort);
+    }
+
+    /**
+     * Fluent method, which sets @code{kafkaConfigurationMap}.
+     *
+     * @param kafkaConfigurationMap kafka configuration
+     * @return StrimziKafkaContainer instance
+     */
+    public StrimziKafkaContainer withKafkaConfigurationMap(final Map<String, String> kafkaConfigurationMap) {
+        this.kafkaConfigurationMap = kafkaConfigurationMap;
+        return this;
+    }
+
+    /**
+     * Fluent method, which sets @code{externalZookeeperConnect}.
+     *
+     * If the broker was created using Kraft, this method throws an {@link IllegalArgumentException}.
+     *
+     * @param externalZookeeperConnect connect string
+     * @return StrimziKafkaContainer instance
+     */
+    public StrimziKafkaContainer withExternalZookeeperConnect(final String externalZookeeperConnect) {
+        if (useKraft) {
+            throw new IllegalArgumentException("Cannot configure an external Zookeeper and use Kraft at the same time");
```

Comment:
IllegalStateException would be better. The problem is that this method is being called _at all_, rather than there being a problem with the argument with which it is being called.